server/athenian/api/controllers/goal_controller.py

```diff
@@ -429,3 +436,40 @@ async def _parse_update_request(
         jira_priorities=jira_priorities,
         jira_issue_types=jira_issue_types,
     )
+
+
+_NUMERIC_THRESHOLD_METRICS = [
+    PullRequestMetricID.PR_REVIEW_COMMENTS_PER_ABOVE_THRESHOLD_RATIO,
+    PullRequestMetricID.PR_SIZE_BELOW_THRESHOLD_RATIO,
+]
+_TIME_DURATION_THRESHOLD_METRICS = [
+    PullRequestMetricID.PR_CYCLE_DEPLOYMENT_TIME_BELOW_THRESHOLD_RATIO,
+    PullRequestMetricID.PR_LEAD_TIME_BELOW_THRESHOLD_RATIO,
+    PullRequestMetricID.PR_OPEN_TIME_BELOW_THRESHOLD_RATIO,
+    PullRequestMetricID.PR_REVIEW_TIME_BELOW_THRESHOLD_RATIO,
+    PullRequestMetricID.PR_WAIT_FIRST_REVIEW_TIME_BELOW_THRESHOLD_RATIO,
+]
+
+
+def _validate_metric_params(goal_metric: str, update_req: GoalUpdateRequest) -> None:
+    if goal_metric in _NUMERIC_THRESHOLD_METRICS:
+        _check_team_goals_threshold_type(goal_metric, update_req, (int, float))
+    elif goal_metric in _TIME_DURATION_THRESHOLD_METRICS:
+        _check_team_goals_threshold_type(goal_metric, update_req, str)
+    else:
+        for tg in update_req.team_goals:
+            if tg.metric_params:
+                raise ResponseError(
+                    InvalidRequestError(".team_goals", f'Invalid team goals: "{tg}"'),
+                )
+
+
+def _check_team_goals_threshold_type(
+    metric: str,
+    update_req: GoalUpdateRequest,
+    types: type | tuple[type, ...],
+) -> None:
+    for tg in [tg for tg in update_req.team_goals if tg.metric_params]:
```

Comment:
too many `tg`-s :smile: 
```suggestion
    for tg in update_req.team_goals:
        if not tg.metric_params:
            continue
```